api/v1/flow/flow.proto

```diff
@@ -137,6 +137,8 @@ message Flow {
     repeated Policy egress_denied_by = 21004;
     // The CiliumNetworkPolicies denying the ingress of the flow.
     repeated Policy ingress_denied_by = 21005;
+    // the set of explicit policy log strings for this flow
+    repeated string policy_log = 21006;
```

Comment:
Two questions:
1. when is this field populated? The only Hubble related change I can see other than this new field introduction is related to the `labels.LabelArrayListString` patch
2. Shouldn't this be part of the `Policy` message?

Reply:
> Two questions:
> 
> 1. when/where is this field populated? The only Hubble related change I can see other than this new field introduction is related to the `labels.LabelArrayListString` patch.

The code populating the field will be submitted in a follow-up PR (that I'm currently working on).

> 2. Shouldn't `policy_log` be part of the `Policy` message?

I thought so too. IIRC this was the case in the initial design @squeed and I sketched out. I'm sure @squeed had a good reason to change it :) but it should probably be documented in the commit message or a comment somewhere.

Reply:
> There is no direct relationship between `egress_allowed_by` (et al.) and `policy_log`, which is intentional.

Ah thank you that wasn't 100% clear to me. I have no problem with that, but then I think it should be more explicit somewhere in the code and/or documentation (or did I miss it?).

Reply:
No, it wasn't clear. I'll add some comments to that effect.